### ISSUE TITLE
How to get rid of parameter? (question or feature request)

### ISSUE THREAD

**Issue by christian-schwaderer** [Unitech/pm2#3282]
Dear all,

I have a special problem with PM2 for which I cannot find a solution. So, maybe this is a question or rather a feature request. Let's see.

For my PM2 process I need a kind of an intial value. So, I start my process like so
`pm2 start my_app.js -x -- myvalue`
`pm2 stave`

At a certain point in my_app.js I'm done with "myvalue" - for every! So, if for any reason, my_app.js should restart I do NOT want to have "myvalue" being a node parameter.
The solution I tried was, restarting the process within itself:
`child_process.execSync('pm2 restart my_app');`

The problem is: I cannot trigger `pm save` afterwards, because everything after pm2 restart my_app is invain, since the process is already restarting.
All I can think of are dirty hacks like: First starting a bash script which sleeps for one second, then triggers `pm2 save`, then, inside my_app.js, trigger the restart. Or put the whole restart thing into a separate bash script or something like that. However, can this be achieved more elegant?
If not: What about this feature? Passing a parameter to `pm2 restart` like "save afterwards"

**Comment by vmarchaud** [comment]
You cannot clear the args with which an application is launched, you should `delete` it and then `start` it again.
If you want to save after a restart simply do `pm2 restart X && pm2 save`

**Comment by christian-schwaderer** [comment]
I cannot start it again from within itself, this is exactly my problem.

This is what I tried.

I created a file namend pm2test.js:

```
const child_process = require('child_process');

setInterval(function() {

 var d = new Date();
 var n = d.toString();
 console.log(n + ' - here with ' + process.argv[2]);

},500);

setTimeout(function() {
   console.log("Will restart now!");
   child_process.execSync('pm2 delete pm2test && pm2 start pm2test.js -x -- newvalue && pm2 save');
},3000);
```

And now started it with  `pm2 start pm2test.js -x -- foo`
What I intended/expected if it worked how I wanted it to be:

- Process prints out 6 times 'here with foo'
- Deletes itself, starts itselfs with a different parameter, saves itself
- Prints out from then on 'here with newvalue'

However, the actual behaviour is:
- Process prints out 6 times 'here with foo'
- Deletes itself and because the process is already stopped/deleted, the lines rest of the command (` && pm2 start pm2test.js -x -- newvalue && pm2 save`) is no more executed.
